File: week1/4/B.py
# https://leetcode.com/problems/the-k-weakest-rows-in-a-matrix/

class Solution:
    def kWeakestRows(mat: list[list[int]], k: int) -> list[int]:
        answers= {}
        for i in range(len(mat)):
            # save dictionaly, index i(key): sumed(value)
            answers[i]= sum(mat[i])
        
        # 
        answers= sorted(answers, key=answers.get)

        return answers[:k]

mat= [[1,1,0,0,0],
 [1,1,1,1,0],
 [1,0,0,0,0],
 [1,1,0,0,0],
 [1,1,1,1,1]]
k= 3
# Output: [2,0,3]
result= Solution.kWeakestRows(mat, k)
print(result)

File: week1/4/B.java
//https://st-lab.tistory.com/261

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

//https://tutorialcup.com/leetcode-solutions/the-k-weakest-rows-in-a-matrix-leetcode-solution.htm
class Solution {
    public static int numOnes(int[] row){

    }

    public static int[] kWeakestRows(int[][] mat, int k) {

    }
    public static void main(String[] args) {
        int [][] arr ={      {1,1,0,0,0 },
                { 1,1,1,1,0 },
                { 1,0,0,0,0 },
                { 1,1,0,0,0 },
                { 1,1,1,1,1 }};
        int k=3;
        int[]ans=kWeakestRows(arr,k);
        System.out.println(Arrays.toString(ans));
    }
}